File: app/src/main/java/ollu/dp/ua/weather_test/model/WeatherData.java
package ollu.dp.ua.weather_test.model;

/**
 * ----
 * Created by Лукащук Олег(master) on 06.03.18.
 */

@SuppressWarnings("ALL")
public class WeatherData {
    public Coord coord;
    public String id;
    public String name;
    public String localName;
    public String cod;
    public Weather[] weather;
    public String base;
    public Main main;
    public int visibility;
    public Wind wind;
    public Clouds clouds;
    public long dt;
    public Sys sys;

    public static class Coord {
        public double lon;
        public double lat;
    }

    public static class Weather {
        public String description;
        public int id;
        public String main;
        public String icon;
    }

    public static class Main {
        public float temp;
        public float temp_min;
        public float temp_max;
        public int pressure;
        public int humidity;
    }

    public static class Wind {
        public float speed;
        public int deg;
    }

    public static class Clouds { //Cloudiness %
        public int all; //%
    }

    public static class Sys {
        public int type;
        public int id;
        public double message;
        public String country;
        public long sunrise;
        public long sunset;
    }
}

File: app/src/main/java/ollu/dp/ua/weather/model/WeatherData.kt
package ollu.dp.ua.weather.model

/**
 * ----
 * Created by Лукащук Олег(master) on 06.03.18.
 */
@Suppress("unused")
class WeatherData {
    var coord: Coord? = null
    var id: String? = null
    var name: String? = null
    var localName: String? = null
    var cod: String? = null
    @JvmField
    var weather: Array<Weather>? = null
    var base: String? = null
    @JvmField
    var main: Main? = null
    var visibility: Int = 0
    var wind: Wind? = null
    var clouds: Clouds? = null
    var dt: Long = 0
    var sys: Sys? = null

    class Coord {
        var lon: Double = 0.toDouble()
        var lat: Double = 0.toDouble()
    }

    class Weather {
        @JvmField
        var description: String? = null
        var id: Int = 0
        var main: String? = null
        var icon: String? = null
    }

    @Suppress("PropertyName")
    class Main {
        var temp: Float = 0.toFloat()
        var temp_min: Float = 0.toFloat()
        var temp_max: Float = 0.toFloat()
        var pressure: Int = 0
        var humidity: Int = 0
    }

    class Wind {
        var speed: Float = 0.toFloat()
        var deg: Int = 0
    }

    class Clouds { //Cloudiness %
        var all: Int = 0 //%
    }

    class Sys {
        var type: Int = 0
        var id: Int = 0
        var message: Double = 0.toDouble()
        var country: String? = null
        var sunrise: Long = 0
        var sunset: Long = 0
    }
}

